includes/admin/class-wc-rest-stripe-orders-controller.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Class WC_REST_Stripe_Orders_Controller
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * REST controller for orders.
+ */
+class WC_REST_Stripe_Orders_Controller extends WC_Stripe_REST_Base_Controller {
+
+	/**
+	 * Endpoint path.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'wc_stripe/orders';
+
+	/**
+	 * Stripe payment gateway.
+	 *
+	 * @var WC_Gateway_Stripe
+	 */
+	private $gateway;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param WC_Gateway_Stripe $gateway Stripe payment gateway.
+	 */
+	public function __construct( WC_Gateway_Stripe $gateway ) {
+		$this->gateway = $gateway;
+	}
+
+	/**
+	 * Configure REST API routes.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<order_id>\d+)/create_customer',
+			[
+				'methods'             => WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'create_customer' ],
+				'permission_callback' => [ $this, 'check_permission' ],
+			]
+		);
+	}
+
+	/**
+	 * Create a Stripe customer for an order if needed, or return existing customer.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function create_customer( $request ) {
+		$order_id = $request['order_id'];
+
+		// Ensure order exists.
+		$order = wc_get_order( $order_id );
+		if ( false === $order || ! ( $order instanceof WC_Order ) ) {
+			return new WP_Error( 'wc_stripe', __( 'Order not found', 'woocommerce-gateway-stripe' ), [ 'status' => 404 ] );
+		}
+
+		// Validate order status before creating customer.
+		$disallowed_order_statuses = apply_filters( 'wc_stripe_create_customer_disallowed_order_statuses', [ 'completed', 'cancelled', 'refunded', 'failed' ] );
```

Comment:
Not something that would block approval, but I was wondering whether `processing` should be included here. I went to the P2 post for details, but couldn't fine anything about statuses. I'm asking because during the regular payment flow, "processing" is set after payment has been processed, so it seems a bit late to be creating the related customer, but not sure if things are different in the work you're doing.

Feel free to adjust or leave as is. Not something that would warrant a review.

Reply:
Thanks for the review, @jorgeatorres! It looks like WCPay's handling of customer creation intentionally includes `processing` as one of the allowed statuses (more details in https://github.com/Automattic/woocommerce-payments/pull/2626), so for consistency I think it's okay to leave this one as-is.